src/wind_energy/BoussinesqBuoyancy.cpp

```diff
@@ -39,15 +41,20 @@ BoussinesqBuoyancy::BoussinesqBuoyancy()
  *  @param vel_forces Forcing source term where Boussinesq term is added
  */
 void BoussinesqBuoyancy::operator()(
+    const int lev,
+    const amrex::MFIter& mfi,
     const amrex::Box& bx,
-    const amrex::Array4<const amrex::Real>& scalars,
+    const FieldState fstate,
     const amrex::Array4<amrex::Real>& vel_forces) const
 {
     const amrex::Real T0 = m_ref_theta;
     const amrex::Real beta = m_beta;
     const amrex::GpuArray<amrex::Real, AMREX_SPACEDIM> gravity{
         m_gravity[0], m_gravity[1], m_gravity[2]};
 
+    FieldState tstate = field_impl::phi_state(fstate);
+    const auto& scalars = m_repo.get_field("temperature", tstate)(lev).const_array(mfi);
```

Comment:
So the idea would be that we would declare a class member 

```C++
// in class header
Field& m_temperature;

BoussinesqBuoyancy()
  : m_repo(repo)
  , m_temperature(repo.get_field("temperature"))
```

So that we can just use the field directly without doing a `get_field` within the `MFIter` loop

Reply:
to do this we'll have to move the physics out of the read parameters and after the declare_fields, but I'm guessing you already have a plan for that too...